specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2023-09-01-preview/apimworkspacediagnostics.json

```diff
@@ -0,0 +1,766 @@
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DiagnosticContract"
+            },
+            "description": "Create parameters."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchOptionalParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Diagnostic was successfully created.",
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DiagnosticContract"
+            }
+          },
+          "200": {
+            "description": "Diagnostic successfully updated",
+            "headers": {
+              "ETag": {
+                "description": "Current entity state version. Should be treated as opaque and used to make conditional HTTP requests.",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DiagnosticContract"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Diagnostic"
+        ],
+        "operationId": "WorkspaceDiagnostic_Update",
+        "description": "Updates the details of the Diagnostic specified by its identifier.",
+        "x-ms-examples": {
+          "ApiManagementUpdateWorkspaceDiagnostic": {
+            "$ref": "./examples/ApiManagementUpdateWorkspaceDiagnostic.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/WorkspaceIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/DiagnosticIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DiagnosticContract"
```

Comment:
>DiagnosticContract [](http://example.com/codeflow?start=55&length=18)

Should this be DiagnosticContractProperties? I don't think top-level properties (other than tags on tracked resources) are supposed to be in the PATCH payload.

Reply:
Actually, I did some more research, and it's fine leaving it as DiagnosticContract if you are using common types definition for the top-level properties, or if you've defined them yourself with the correct mutability. Shouldn't be an issue. In fact, I think we are recommending that approach now.

Reply:
I rollbacked the changes there as DiagnosticContract is a way to go here